tests/integration/security/util/reachability/context.go

```diff
@@ -166,7 +165,9 @@ func Run(testCases []TestCase, t framework.TestContext, apps *util.EchoDeploymen
 								// Set the target on the call options.
 								opts.To = to
 								opts.Count = callCount
-
+								if c.ExpectReached != nil && !c.ExpectReached(opts) {
```

Comment:
A bit confused by behavior of ExpectReached and the existing ExpectDestinations.

ExpectDestinations sounds like its changing the test _assertion_ - but its not... its actually changing where we send. It seems like this should be `FilterDestinations` (or just `Include` and just add `to` to that). This is existing code, but seems related

ExpectReached sounds like it is changing the _assertion_ as well (actually sounds identical to ExpectDestinations to me), but its not -- its skipping tests. Should this either just use `Include` or make it so we don't skip the test but rather run it and do not assert that all destinations are reached?

Reply:
I thought ExpectDestinations skipped tests and ExpectedReached changed assertions. ExpectDestinations can probably just be handled in `Include`. ExpectReached has to have the `call opts` because expectations change for things like per-port settings.

Reply:
Oh, misread your comment. We're on the same page.

Reply:
I agree with you John, I updated the code and added the logic of `ExpectedReached` inside `Include`.